ledger/participant-integration-api/src/main/resources/db/migration/oracle/V1__Init.sql

```diff
@@ -0,0 +1,325 @@
+    tree_event_witnesses          VARCHAR_ARRAY                                   not null,
+
+    event_sequential_id           NUMBER GENERATED BY DEFAULT ON NULL AS IDENTITY not null,
+    create_argument_compression   NUMBER,
+    create_key_value_compression  NUMBER,
+    exercise_argument_compression NUMBER,
+    exercise_result_compression   NUMBER
+);
+
+-- support ordering by offset and transaction, ready for serving via the Ledger API
+create index participant_events_offset_txn_node_idx on participant_events (event_offset, transaction_id, node_index);
+
+-- support looking up a create event by the identifier of the contract it created, so that
+-- consuming exercise events can use it to set the value of create_consumed_at
+create index participant_events_contract_idx on participant_events (contract_id);
+
+-- support requests of transactions by transaction_id
+create index participant_events_transaction_idx on participant_events (transaction_id);
+
+-- support filtering by template
+create index participant_events_template_ids on participant_events (template_id);
+
+-- 4. create a new index involving event_sequential_id
+create index participant_events_event_sequential_id on participant_events (event_sequential_id);
+
+-- 5. we need this index to convert event_offset to event_sequential_id
+create index participant_events_event_offset on participant_events (event_offset);
+
+-- TODO https://github.com/digital-asset/daml/issues/9493
+-- create index participant_events_flat_event_witnesses_idx on participant_events (flat_event_witnesses);
+-- create index participant_events_tree_event_witnesses_idx on participant_events (tree_event_witnesses);
+
+
+---------------------------------------------------------------------------------------------------
+-- V26: Contracts new schema
+--
+-- Used for interpretation and validation by the DAML engine
+---------------------------------------------------------------------------------------------------
+
+-- contains all active and divulged contracts
+create table participant_contracts
+(
+    contract_id                  NVARCHAR2(1000) primary key not null,
+
+    -- template_id and create_argument need to be nullable to leave this fields empty as part of this migration
+    -- a second (java) migration will fill them up
+    -- a third (sql) migration will make them non-nullable, as it's supposed to always be populated
+    template_id                  NVARCHAR2(1000)             not null,
+    create_argument              BLOB                        not null,
+
+    -- the following fields are null for divulged contracts
+    create_stakeholders          VARCHAR_ARRAY,
+    create_key_hash              VARCHAR2(4000),
+    create_ledger_effective_time TIMESTAMP,
+    create_argument_compression  SMALLINT
+);
+
+-- support looking up a contract by key
+create unique index participant_contracts_idx on participant_contracts (create_key_hash);
```

Comment:
fwiw, in the canton fork, we will end up turning `participant_contracts_idx` to be non-unique.

Reply:
yeah explicitly named to make the indices easier to manage -- with oracle the naming of anonymous indices is a bit inscrutable